src/main/kotlin/io/libp2p/pubsub/gossip/GossipScore.kt

```diff
@@ -116,6 +117,14 @@ class GossipScore(
 
     private fun isInMesh(peer: P2PService.PeerHandler, topic: Topic) = getTopicScores(peer, topic).inMesh()
 
+    fun updateTopicParams(topicScoreParams: Map<String, GossipTopicScoreParams>) {
+        executor.execute {
```

Comment:
nit: does it make sense just to make `topicsScoreParams.topicParams` map synchronized and call `setTopicParams` directly?

Reply:
Not sure - if we moved to a synchronized map, I think we can get cases where `calcTopicScore` performs part of the calculation with the old topic params and part of the calculation with new topic params.  Maybe that's okay?  But seems unexpected.

Reply:
You are right it's better to update them atomically. Let's leave the `executor` then